Clean up catalog configs and README

yymao opened the pull request:
This PR

- adds the `deprecated` flags to `dr6a`, `dr6b`, and `dr6c` catalogs, prompting users to switch to `dr6_wfd` (Note: all the `deprecated` catalogs are still accessible, but a warning will be printed).
- creates `_with_all_addons` configs for `dr6_wfd` and `dr2` object catalogs that include all available addons (not for `dr3` because the metacal are missing tracts)
- fixes minor issues in catalog descriptions and settings.
- updates the README. 

Here's a screenshot of the README. You can find the README [here](https://github.com/LSSTDESC/gcr-catalogs/blob/u/yymao/clean-up-catalog-configs/README.md#dc2-runs-data-products-and-add-ons)

![image](https://user-images.githubusercontent.com/3792659/103010980-2c954600-4507-11eb-8e13-d5111f83c98c.png)

yymao commented:
For the DR3 catalogs, based a slack discussion, we opted to list the catalogs with `dr3a` in their names instead of those with `dr3`, so that it is clear that these catalogs are preview. Here's the updated readme of that section:

![image](https://user-images.githubusercontent.com/3792659/103014714-3c178d80-450d-11eb-8159-3053498ad716.png)

Here's what a user would see if they list all the "default" catalogs that contains "run2.2i" in the names:

```python
>>> import GCRCatalogs
>>> GCRCatalogs.get_available_catalogs(names_only=True, name_contains="run2.2i")
['dc2_object_run2.2i_dr2',
 'dc2_object_run2.2i_dr2_with_all_addons',
 'dc2_object_run2.2i_dr3a',
 'dc2_object_run2.2i_dr3a_with_metacal',
 'dc2_object_run2.2i_dr3a_with_photoz',
 'dc2_object_run2.2i_dr6_wfd',
 'dc2_object_run2.2i_dr6_wfd_with_all_addons',
 'dc2_truth_match_run2.2i_dr6_wfd',
 'dc2_truth_run2.2i_galaxy_truth_summary',
 'dc2_truth_run2.2i_sn_truth_summary',
 'dc2_truth_run2.2i_sn_variability_truth',
 'dc2_truth_run2.2i_star_lc_stats',
 'dc2_truth_run2.2i_star_truth_summary',
 'dc2_truth_run2.2i_star_variability_truth']
```

heather999 commented:
I'm not as sure we want to revert to `dr3a` and remove `dr3`.  We have been referring to `dr3` for months and it is mentioned as "dr3" on Confluence pages like https://confluence.slac.stanford.edu/display/LSSTDESC/DESC+DRP+DC2+Run2.2i+Processing and https://confluence.slac.stanford.edu/display/LSSTDESC/DC2+Data+Product+Overview 
Should we go back and update those pages and similarly the alias in [desc-dc2-dm-data](https://github.com/LSSTDESC/desc-dc2-dm-data/blob/master/desc_dc2_dm_data/repos.py)?  I think it's perfectly valid to indicate that dr3 is preview or something along those lines.

yymao commented:
Currently `dr3` and `dr3a` are exactly the same, and both would work in GCRCatalogs. 

The question is really just which one we put in the default list and README.  I guess it depends on what we want to stress and I don't really have strong feelings here. If we want to stress that this is still a preview, using `dr3a` in the default list and README seems reasonable. If we want to stress that this is the "most up-to-date" DR3, then using just `dr3` seems better.

katrinheitmann commented:
I think making it very clear that it is a preview is important. It is incomplete and it was only called DR3 by mistake in that we did not realized at first that it had the hole. So I think we should call it DR3a and if we need to change some things elsewhere we should do that. Otherwise it might really confuse people.

heather999 commented:
Ok - dr3a should be in the README - and I can update Confluence pages to offer some consistency and also look at desc-dc2-dm-data.

yymao commented:
OK, this PR is now ready for review, but it's not easy to review given how much renaming we have done. 

In short, for Run 2.2i, we settle with the names `dr6`, dr2_wfd`, and `dr3a`.  So the catalogs in the default list include:

- `dc2_object_run2.2i_dr6` (currently points to `dc2_object_run2.2i_dr6_v1`
- `dc2_object_run2.2i_dr6_with_addons` (currently points to `dc2_object_run2.2i_dr6_v1_with_addons`)
- `dc2_object_run2.2i_dr2_wfd` (currently points to `dc2_object_run2.2i_dr2_wfd_v1`
- `dc2_object_run2.2i_dr2_wfd_with_addons` (currently points to `dc2_object_run2.2i_dr2_wfd_v1_with_addons`)
- `dc2_object_run2.2i_dr3a` (currently the same as `dc2_object_run2.2i_dr3`)
- `dc2_object_run2.2i_dr3a_with_metacal` (currently points to `dc2_object_run2.2i_dr3_with_metacal`)
- `dc2_object_run2.2i_dr3a_with_photoz` (currently points to `dc2_object_run2.2i_dr3_with_photoz`)

These are also the catalogs listed in README. Note that the catalogs in parenthesis are also accessible and not deprecated. Here's an example: 

```python
>>> GCRCatalogs.get_available_catalogs(names_only=True, name_contains="object_run2.2i")
['dc2_object_run2.2i_dr2_wfd',
 'dc2_object_run2.2i_dr2_wfd_with_addons',
 'dc2_object_run2.2i_dr3a',
 'dc2_object_run2.2i_dr3a_with_metacal',
 'dc2_object_run2.2i_dr3a_with_photoz',
 'dc2_object_run2.2i_dr6',
 'dc2_object_run2.2i_dr6_with_addons']

>>> GCRCatalogs.get_available_catalogs(include_default_only=False, names_only=True, name_contains="object_run2.2i")
['dc2_object_run2.2i_dr2_wfd',
 'dc2_object_run2.2i_dr2_wfd_v1',
 'dc2_object_run2.2i_dr2_wfd_v1_with_addons',
 'dc2_object_run2.2i_dr2_wfd_with_addons',
 'dc2_object_run2.2i_dr3',
 'dc2_object_run2.2i_dr3_with_metacal',
 'dc2_object_run2.2i_dr3_with_photoz',
 'dc2_object_run2.2i_dr3a',
 'dc2_object_run2.2i_dr3a_with_metacal',
 'dc2_object_run2.2i_dr3a_with_photoz',
 'dc2_object_run2.2i_dr6',
 'dc2_object_run2.2i_dr6_v1',
 'dc2_object_run2.2i_dr6_v1_with_addons',
 'dc2_object_run2.2i_dr6_with_addons',
 'dc2_object_run2.2i_dr6a_with_photoz']
```

With the exception of `dc2_object_run2.2i_dr6a_with_photoz` (because we don't have photo-z for dr6 yet), we deprecate all catalogs that have `dr6_wfd`, `dr6a`, `dr6b`, `dr6c` in their names:

```python
>>> GCRCatalogs.get_available_catalogs(include_default_only=False, names_only=True, name_contains="object_run2.2i", include_deprecated=True, additional_conditions=[lambda c: c.is_deprecated])
['dc2_object_run2.2i_dr6_wfd',
 'dc2_object_run2.2i_dr6_wfd_matched_addon',
 'dc2_object_run2.2i_dr6_wfd_v1',
 'dc2_object_run2.2i_dr6_wfd_v1_with_metacal',
 'dc2_object_run2.2i_dr6_wfd_with_metacal',
 'dc2_object_run2.2i_dr6a',
 'dc2_object_run2.2i_dr6a_matched_addon',
 'dc2_object_run2.2i_dr6a_with_metacal',
 'dc2_object_run2.2i_dr6b',
 'dc2_object_run2.2i_dr6c',
 'dc2_object_run2.2i_dr6c_matched_addon',
 'dc2_object_run2.2i_dr6c_with_metacal']
```

Users who try to load these catalogs will still be able to, but they will be prompted to switch to `dc2_object_run2.2i_dr6` or `dc2_object_run2.2i_dr6_with_addons` if they load `dr6x`, and to `dc2_object_run2.2i_dr6_v1` or `dc2_object_run2.2i_dr6_v1_with_addons` if they load `dr6_wfd`. 

@heather999, @JoanneBogart, if you can give this a review that'd be great. As I mentioned earlier, it's quite difficult to track the changes given the amount of renaming. I'd be happy to answer questions if you have any.

For completeness, the rendered version of README is here:

![image](https://user-images.githubusercontent.com/3792659/103041229-ca0e6b00-4543-11eb-840f-379755cf24af.png)

JoanneBogart commented on GCRCatalogs/catalog_configs/dc2_object_run2.2i_dr6_wfd_with_metacal.yaml:
```diff
@@ -1,2 +1,2 @@
-alias: dc2_object_run2.2i_dr6_wfd_v1_with_metacal
-include_in_default_catalog_list: true
+alias: dc2_object_run2.2i_dr6_v1_with_addons
+deprecated: Use dc2_object_run2.2i_dr6_v1_with_addons
```
```suggestion
alias: dc2_object_run2.2i_dr6_with_addons
include_in_default_catalog_list: true
deprecated: Use dc2_object_run2.2i_dr6_with_addons
```

JoanneBogart reviewed (changes requested):
Thanks for tackling this, @yymao!  There are a couple things which I believe must be addressed (see comment for old `dc2_object_run2.2i_dr2_with_truth_match.yaml` and issue about  `deprecated` message for composite catalog `dc2_object_run2.2i_dr6_wfd_matched_addon`)   but most comments are suggestions that fall into a few categories:
* For composites, don't use deprecated catalog - e.g., name with `dr6` and `wfd` in it if there is an equivalent not-deprecated alternative
* If catalog name doesn't include v1, certain internal references - e.g. when suggesting an alternative in the `deprecated` message - shouldn't either
* Small changes to text in the README.  I would also suggest minor changes to the first paragraph under `Use GCRCatalogs at NERSC` but that is not text you changed in this PR.

yymao commented on GCRCatalogs/catalog_configs/dc2_object_run2.2i_dr3_tract3830.yaml:
```diff
@@ -1,3 +0,0 @@
-based_on: dc2_object_run2.2i_dr3
-tract: 3830
```
I think this was a mis-addition from the beginning. In Run 1.x and early Run 2.1, when we still used hdf5 for object catalogs, the reader needs to read the metadata of all files at load time, and hence a single-tract config can speed up the catalog load time for quick test. 

In the new parquet format, the reader doesn't not read all the files in load time, and single-tract config would not speed up anything. The user can and should use native filter if they want to load a single tract of data.

yymao commented:
Thanks @JoanneBogart --- I am a little bit unsure about suggesting `dr6` rather than `dr6_v1` when the user asks for `dr6_wfd`. For `dr6`, the postfix `_wfd` does not mean the same thing as in `dr2_wfd`. For `dr6`, that postfix should have been called `_ddf_region_excluded`. We now use `_v1` to indicate such exclusion (that is, it is possible that we will have a v2 that includes the DDF region). This is why I mapped `dr6_wfd` to `dr6_v1` instead of `dr6`.

JoanneBogart commented on GCRCatalogs/catalog_configs/dc2_object_run2.2i_dr6_wfd_matched_addon.yaml:
```diff
@@ -1,13 +1,20 @@
-creators: ['Javi Sanchez', 'Eve Kovacs']
+based_on: dc2_object_run2.2i_dr6_v1_with_addons
 catalogs:
   - catalog_name: dc2_object_run2.2i_dr6_wfd_v1
```
Maybe a separate issue from this PR, but it seems that the `deprecated` key is ignored for catalogs which are either composites or use `based_on`

While investigating I came across two catalogs whose `deprecated` messages reference catalogs which no longer exist:
`dc2_object_run2.1i_dr1_tract3830`
`dc2_object_run2.1i_dr4_tract3830`

yymao replied:
The `deprecated` key indeed only works if it appears in the top level. This is by design rather than a bug. The reason behind is to make its behavior similar to `include_in_default_catalog_list`. 

The two `_tract3830` catalogs should not exist in the first place too. Maybe instead of deleting it I should deprecate them instead.

JoanneBogart replied:
It seems to be the opposite  - it doesn't work at the top level, but I did see messages when referenced deprecated catalogs internally

yymao commented:
@JoanneBogart  I think I addressed all of your suggestions except for the `dr6_wfd` -> `dr6` due to the reason I mentioned above. 

You mentioned that you have further suggestions on the README, do you want to comment here so that I can add it to the PR or do you want to open a new PR?

JoanneBogart commented:
I believe I've resolved all the conversations having to do with `dr6_wfd` -> `dr6`; your point is a good one.

Concerning the README, I suggest
* Change title to **Using GCRCatalogs at NERSC**
* Omit first sentence (redundant)
* Modify next sentence to read "All catalogs available in `GCRCatalogs` are physically located at NERSC"

There was one other issue which I thought I commented on but can no longer find my comment:  the lack of a `deprecated` message in certain circumstances (composites, `based_on`), even though the `deprecated` keyword is used in the config.

yymao commented:
@JoanneBogart re: your comment on README, do you mean to change the overall title of the README to "Using GCRCatalogs at NERSC"? And why is the first sentence "`GCRCatalogs` is a Python package that serves as a repository of various galaxy catalogs and sky catalogs for the LSST Dark Energy Science Collaboration (DESC)." redundant? I must miss something here..

yymao commented:
Re: `deprecated` -- yes, I replied [here](https://github.com/LSSTDESC/gcr-catalogs/pull/528#discussion_r549456901).

JoanneBogart commented:
> re: your comment on README, do you mean to change the overall title of the README to "Using GCRCatalogs at NERSC"? And why is the first sentence "GCRCatalogs is a Python package that serves as a repository of various galaxy catalogs and sky catalogs for the LSST Dark Energy Science Collaboration (DESC)." redundant? I must miss something here..

Sorry; I wasn't clear.   My comments all refer to the section **Use GCRCatalogs at NERSC**
That's the heading that should be changed, the first sentence removed, etc.

yymao commented:
Ah, thanks for the clarification! I've updated the README.

yymao commented:
It turns out that `DeprecationWarning` would not show by default, so I am changing it to a regular UserWarning.

JoanneBogart commented on README.md:
```diff
@@ -142,11 +142,11 @@ Note: DR3 processing is not fully completed; a few tracts are missing. Here `dr3
+All catalogs available in `GCRCatalogs` are physically located at NERSC
+(most are mirrored at IN2P3-CC as well).
+At NERSC, you need to be in the `lsst` user group to access them.
 You can find instructions about getting NERSC account and joining `lsst` group
```
Line 150 should read
"You can find instructions about getting a NERSC account and joining `lsst` group"

That is, insert "a" before "NERSC"  (should have caught this the first time; sorry)

yymao replied:
Fixed, thanks!

JoanneBogart reviewed (dismissed):
I recommend one last teeny change to the README; otherwise looks good!

JoanneBogart reviewed (approved):
LGTM.  Thanks!
